packages/build-tools/src/steps/functions/internalRestoreCache.ts

```diff
@@ -0,0 +1,32 @@
+import { BuildFunction, BuildStepInput, BuildStepInputValueTypeName } from '@expo/steps';
+import { Platform } from '@expo/eas-build-job';
+
+import { restoreCcacheAsync } from './restoreCache';
+
+export function createInternalRestoreCacheFunction(): BuildFunction {
+  return new BuildFunction({
+    namespace: 'eas',
+    id: 'restore_build_cache',
+    name: 'Restore Cache',
+    inputProviders: [
+      BuildStepInput.createProvider({
+        id: 'platform',
+        required: true,
```

Comment:
Can we do sth like in https://github.com/expo/eas-build/blob/d0e2a0e4dd02264528236e7ff8279a8863aee277/packages/build-tools/src/steps/functions/calculateEASUpdateRuntimeVersion.ts#L49-L50? optional parameter, inferred from job context if possible